src/frontend/data/sideEffects/pollForTrack/index.spec.tsx

```diff
@@ -0,0 +1,104 @@
+import { wait } from '@testing-library/react';
+import fetchMock from 'fetch-mock';
+
+import { pollForTrack } from '.';
+import { requestStatus } from '../../../types/api';
+import { modelName } from '../../../types/models';
+import { report } from '../../../utils/errors/report';
+
+jest.mock('../../../utils/errors/report', () => ({ report: jest.fn() }));
+
+describe('sideEffects/pollForTrack', () => {
+  const dispatch = jest.fn();
+  const doPollForTrack = pollForTrack(dispatch, 'some token');
+
+  jest.useFakeTimers();
+
+  afterEach(fetchMock.restore);
+
+  it('polls the track, backing off until it is ready and resolves with a success', async () => {
+    fetchMock.mock(
+      '/api/videos/42/',
+      JSON.stringify({ is_ready_to_play: false }),
+      { method: 'GET' },
+    );
+    const promise = doPollForTrack(modelName.VIDEOS, '42', 15);
+
+    expect(
+      fetchMock.calls('/api/videos/42/', { method: 'GET' }).length,
+    ).toEqual(1);
+
+    await wait();
+    jest.advanceTimersByTime(2 * 15 * 1000 + 200);
+    await wait();
+
+    expect(
+      fetchMock.calls('/api/videos/42/', { method: 'GET' }).length,
+    ).toEqual(2);
+
+    await wait();
+    jest.advanceTimersByTime(2 * 3 * 15 * 1000 + 200);
+    await wait();
+
+    expect(
+      fetchMock.calls('/api/videos/42/', { method: 'GET' }).length,
+    ).toEqual(3);
+
+    fetchMock.mock(
+      '/api/videos/42/',
+      JSON.stringify({ is_ready_to_play: true }),
+      { method: 'GET', overwriteRoutes: true },
+    );
+
+    await wait();
+    jest.advanceTimersByTime(2 * 3 * 4 * 15 * 1000 + 200);
+    await wait();
+
+    expect(
+      fetchMock.calls('/api/videos/42/', { method: 'GET' }).length,
+    ).toEqual(4);
+    expect(await promise).toEqual(requestStatus.SUCCESS);
+
+    await wait();
+    jest.advanceTimersByTime(2 * 3 * 4 * 5 * 15 * 1000 + 200);
+    await wait();
+
+    // No new calls have been issued
+    expect(
+      fetchMock.calls('/api/videos/42/', { method: 'GET' }).length,
+    ).toEqual(4);
+  });
+
+  it('resolves with a failure and reports it when it fails to poll the track', async () => {
+    fetchMock.mock(
+      '/api/videos/42/',
+      JSON.stringify({ is_ready_to_play: false }),
+      { method: 'GET' },
+    );
+    const promise = doPollForTrack(modelName.VIDEOS, '42', 15);
+
+    // The first call was successful
+    expect(
+      fetchMock.calls('/api/videos/42/', { method: 'GET' }).length,
+    ).toEqual(1);
+
+    fetchMock.mock(
+      '/api/videos/42/',
+      { throws: new Error('Failed to get the track') },
+      {
+        method: 'GET',
+        overwriteRoutes: true,
+      },
+    );
+
+    await wait();
+    jest.advanceTimersByTime(2 * 15 * 1000 + 200);
+    await wait();
```

Comment:
I'm not sure these two `wait` are needed.

Reply:
Without these the request is never triggered as the promises are never resolved